File: HospitalDuty.Domain/Entities/Hospital.cs
using System;

namespace HospitalDuty.Domain.Entities;

public class Hospital
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string Name { get; set; } = string.Empty;
    public string Phone { get; set; } = string.Empty;
    public string District { get; set; } = string.Empty;
    public string City { get; set; } = string.Empty;
    public string Address { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Website { get; set; } = string.Empty;
    public Guid? DirectorId { get; set; }
    public Employee? Director { get; set; }
    public ICollection<Department> Departments { get; set; } = new List<Department>();
    public ICollection<Employee> Employees { get; set; } = new List<Employee>();
    public ICollection<Shift> Shifts { get; set; } = new List<Shift>();
}

File: client/src/types/hospital.ts
export interface Hospital {
  id: string;
  name: string;
  phone: string;
  district: string;
  city: string;
  address: string;
  email: string;
  website: string;
  directorId?: string;
  director?: {
    id: string;
    firstName: string;
    lastName: string;
    fullName?: string;
    email?: string;
    phoneNumber?: string;
    roles?: string[];
    departmentId?: string;
    hospitalId?: string;
    department?: string;
  };
}

export interface CreateHospitalRequest {
  name: string;
  phone: string;
  district: string;
  city: string;
  address: string;
  email: string;
  website: string;
  directorId?: string;
}

export interface UpdateHospitalRequest {
  name: string;
  phone: string;
  district: string;
  city: string;
  address: string;
  email: string;
  website: string;
  directorId?: string;
}
